boltdesignsystem/bolt
# [KEEP OPEN] Video Component Refactor

## Jira

http://vjira2:8080/browse/BDS-539

## Summary

Refactors the existing Video web component + moves rendering over to use lit-html.

## Details

- Convert `Video` and `VideoMeta` from `preact` to `lit-html`
- Pass state to `VideoMeta` with `context`
- Pull in `case-change` library (from Image PR #1032) and convert schema data to camelCase before validation
- Update schema descriptions
- Add props for `no-meta`, `no-meta-title`, `no-controls`
- Update Twig and JS to use new `bolt-ratio` API
- Add `env` to global JS
- Update descriptions on all Video demo pages for consistency
- Add Web Components demo page
- Reorganize sub-component into separate directory

## How to test

- Verify branch will run locally without any issues.

- Review changed files, especially:
  - [video.schema.yml](https://github.com/bolt-design-system/bolt/pull/1099/files#diff-76f3c24c131bbe0a205044daf1331c39)
  - [video.twig](https://github.com/bolt-design-system/bolt/pull/1099/files#diff-280078801b494cd920169c817e93b25b)
  - [_video-tag.twig](https://github.com/bolt-design-system/bolt/pull/1099/files#diff-f7f3e15acb68a499a6a31f58c1ae9f25)
  - [video.js](https://github.com/bolt-design-system/bolt/pull/1099/files#diff-b702e2e01d48047fe2165451870a00f5)
  - [Meta/VideoMeta.js](https://github.com/bolt-design-system/bolt/pull/1099/files#diff-d41770568a84919c935b04f812f9f24a)
  - [core/renderers/bolt-base.js](https://github.com/bolt-design-system/bolt/pull/1099/files#diff-d45fd62db4da766d397ba2b3eaae5990)
  - [core/utils/dataset-to-object.js](https://github.com/bolt-design-system/bolt/pull/1099/files#diff-8cf4258eaf9320ea5e7eda42b31329f7)

- Review new [Web Component demo page](https://feature-convert-video-to-web-component.boltdesignsystem.com/pattern-lab/?p=components-video-web-component) for content and functionality.

- Review [Video view all page](https://feature-convert-video-to-web-component.boltdesignsystem.com/pattern-lab/?p=viewall-components-video) for content and functionality. Descriptions have been updated.
